### ISSUE TITLE
AbsoluteSizeSpan doesn't respect dip=true

### ISSUE THREAD

**Issue by vinc3m1** [facebook/litho#284]
- [x] I have searched [existing issues](https://github.com/facebook/litho/issues) and this is not a duplicate

## Version
0.8.0

## Issues and Steps to Reproduce
Setting text wrapped with an `AbsoluteSizeSpan(sizeDip, true)` results in `sizeDip` always being used as `px` instead of `dip`.

I compared this with a regular TextView hosting the same content and the Litho text is way smaller. Similarly, if I do the work myself of multiplying by `resources.getDisplayMetrics().scaledDensity` then it will the TextView.

Basically, toggling the second parameter `dip=true/false` doesn't do anything in the case of Litho.

## Expected Behavior
`dip=true` in [`AbsoluteSizeSpan`](https://developer.android.com/reference/android/text/style/AbsoluteSizeSpan.html) should be respected.

## Link to Code
https://github.com/vinc3m1/litho/commit/b660222662fc600a62a7783f5af0b1ac68024db3

**Comment by vinc3m1** [comment]
Updated with bug example. Also, this could be a bug in TextLayoutBuilder as well, didn't have time to dig in and see where the root cause was yet.

**Comment by pavlospt** [comment]
@vinc3m1 great, thanks! Will try to investigate as well!

**Comment by pavlospt** [comment]
@vinc3m1 Apparently it seems that it respects the `dip=true` it just does not scale the way someone would expect it to! You can check this if you increase the value of `size` in `AbsoluteSizeSpan`. Still trying to debug why `TextLayoutBuilder` does not measure the appropriate size. Will update if I have anything new!

**Comment by vinc3m1** [comment]
It always treats the size as px, even if `dip=true`. So yes, the size will change, but always as `px`, that's why the regular TextView is so much bigger, because it's multiplied by density.

**Comment by vinc3m1** [comment]
Put another way, if you change `dip=false`, the TextView implementation will change, but the Litho one will not.

**Comment by pavlospt** [comment]
Yeap, that's what I ended up with as well! Attaching a screenshot: 
![24197319_10214722912605863_834870601_o](https://user-images.githubusercontent.com/1070871/33344690-85eeb3aa-d492-11e7-930b-f93bc0c2c500.jpg)
 
That's an `AbsoluteSizeSpan` of 96 `dip` with a `Text` using `textSizeSp(24)` and a `TextView` using `setTextSize(TypedValue.COMPLEX_UNIT_SP, 24)` as well. 

Not sure when `TextLayoutBuilder.setTextSize(int size)` is called on `TextSpec`'s lifecycle! Also, not sure if it needs a special case for `AbsoluteSizeSpan`s like there is one for `ClickableSpan`s!

Maybe Litho team can give us some guidance!

**Comment by pavlospt** [comment]
Is there any update on this :) ?

**Comment by IanChilds** [comment]
It looks like this is something that should be supported by TextLayoutBuilder, but isn't?

**Comment by IanChilds** [comment]
I think we should probably try to fix it at that level? @xiphirx, what do you think?

**Comment by pavlospt** [comment]
To add some more info, I noticed that the measured width was the same for different values on the `AbsoluteSizeSpan`. Not sure if it helps!

**Comment by xiphirx** [comment]
Just ran a simple test with just TextLayoutBuilder and it does indeed seem like the issue is there, so yes we'll have to fix it there.
